code-commit-list-functions.js

```diff
@@ -0,0 +1,60 @@
+const payloadFunctions = require("./payload-functions");
+const { fetchRecursively, splitArrayIntoChunks } = require("./helpers");
+
+async function listProjects(codeBuildClient, params) {
+  const payload = payloadFunctions.prepareListProjectsPayload(params);
+
+  const projects = await fetchRecursively(
+    codeBuildClient,
+    {
+      methodName: "listProjects",
+      outputDataPath: "projects",
+    },
+    payload,
+  ).catch((error) => {
```

Comment:
Don't mix promise with async/await approach. The latter is what we agreed to use at plugins and its way of handling errors is a try-catch block.